packages/coinstac-docker-manager/test/index.test.js

```diff
@@ -0,0 +1,348 @@
+});
+
+test.serial.before('pullImagesFromList', async (t) => {
+  sinon.stub(docker, 'pull').yields(null, 'stream');
+
+  const compList = COMPUTATIONS.map(comp => comp.img);
+  const images = await pullImagesFromList(compList);
+
+  t.is(compList.length, images.length);
+
+  docker.pull.restore();
+});
+
+test.serial.before('pruneImages', async (t) => {
+  sinon.stub(docker, 'listImages').yields(null, ALL_IMAGES);
+
+  const getImageSuccessCallback = { remove: sinon.stub().yields(null, 'success') };
+  const getImageFailCallback = { remove: sinon.stub().yields('error') };
+
+  sinon.stub(docker, 'getImage')
+    .onFirstCall()
+    .returns(getImageFailCallback)
+    .returns(getImageSuccessCallback);
+
+  const res1 = await pruneImages();
+  t.is(res1.filter(elem => elem === 'success').length, 9);
+
+  docker.listImages.restore();
+
+  sinon.stub(docker, 'listImages').yields('error', null);
+  const res2 = await pruneImages();
+  t.is(res2, undefined);
+
+  docker.listImages.restore();
+  docker.getImage.restore();
+});
+
+test('removeImage', async (t) => {
+  const getImageSuccessCallback = { remove: sinon.stub() };
+  sinon.stub(docker, 'getImage').returns(getImageSuccessCallback);
+  const image = COMPUTATIONS[0].img;
+
+  await removeImage(image);
+  t.pass();
+
+  docker.getImage.restore();
+});
+
+test('setLogger', async (t) => {
+  const logger = winston.loggers.add('coinstac-main', {
+    level: 'silly',
+    transports: [new Console()],
+  });
+  setLogger(logger);
+
+  t.pass();
+});
+
+test('startService1', async (t) => {
+  sinon.stub(portscanner, 'findAPortNotInUse').returns(getRandomPort());
+  sinon.stub(docker, 'createContainer').returns({
+    start: sinon.stub(),
+    stop: sinon.stub(),
+    inspect: sinon.stub().yields(null, { State: { Running: true } }),
+  });
+
+  const opts = {
+    docker: { CMD: [] },
+  };
+
+  await startService('service-1', 'test-1', opts);
+  t.pass();
+
+  portscanner.findAPortNotInUse.restore();
+  docker.createContainer.restore();
+});
+
+test.serial.before('startService2', async (t) => {
+  sinon.stub(portscanner, 'findAPortNotInUse').returns(getRandomPort());
+  sinon.stub(docker, 'createContainer').returns({
+    start: sinon.stub(),
+    stop: sinon.stub(),
+    inspect: sinon.stub().yields('error', null),
+  });
+
+  const opts = {
+    docker: { CMD: [] },
+  };
+
+  await startService('service-2', 'test-2', opts);
+  t.pass();
+
+  portscanner.findAPortNotInUse.restore();
+  docker.createContainer.restore();
+});
+
+test.serial.before('startService3', async (t) => {
+  sinon.stub(portscanner, 'findAPortNotInUse').returns(getRandomPort());
+  sinon.stub(docker, 'createContainer').returns({
+    start: sinon.stub(),
+    stop: sinon.stub(),
+    inspect: sinon.stub().yields(null, { State: { Running: true } }),
+  });
+
+  const opts = {
+    docker: { CMD: [] },
+  };
+
+  await startService('service-3', 'test-3', opts);
+  await startService('service-3', 'test-3', opts);
+  t.pass();
```

Comment:
Again here, what is this checking? Having multiple clients is good to hit come code conditions on how `startService` gives you a service function, but why 3+ different tests for it?